packages/react-priority-overflow/src/useOverflowMenu.ts

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { useId, useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+import { useOverflowContext } from './overflowContext';
+import { useOverflowCount } from './useOverflowCount';
+import { DATA_OVERFLOW_MENU } from './constants';
+
+export function useOverflowMenu<TElement extends HTMLElement>(id?: string) {
+  const elementId = useId('overflow-menu', id);
+  const overflowCount = useOverflowCount();
+  const registerItem = useOverflowContext(v => v.registerItem);
+  const updateOverflow = useOverflowContext(v => v.updateOverflow);
+  const ref = React.useRef<TElement>(null);
+  const isOverflowing = overflowCount > 0;
+
+  useIsomorphicLayoutEffect(() => {
+    let deregisterItem: () => void = () => null;
+    const element = ref.current;
+    if (element) {
+      deregisterItem = registerItem({
+        element,
+        id: elementId,
+        priority: Infinity,
+      });
+      element.setAttribute(DATA_OVERFLOW_MENU, '');
+    }
+
+    return () => {
+      element?.removeAttribute(DATA_OVERFLOW_MENU);
+      deregisterItem();
+    };
```

Comment:
```suggestion
    const element = ref.current;
    
    if (element) {  
      const unregisterItem = registerItem({
        element,
        id: elementId,
        priority: Infinity,
      });
      element.setAttribute(DATA_OVERFLOW_MENU, '');
      
      return () => {
        unregisterItem();
        element.removeAttribute(DATA_OVERFLOW_MENU);
      };
    }
```

?

Reply:
having branching returns in an effect is an interesting pattern, I haven't seen it before